service/history/events/cache.go

```diff
@@ -48,20 +48,23 @@ type (
 			runID string,
 			firstEventID int64,
 			eventID int64,
+			version int64,
 			branchToken []byte,
 		) (*historypb.HistoryEvent, error)
 		PutEvent(
 			namespaceID string,
 			workflowID string,
 			runID string,
 			eventID int64,
+			version int64,
```

Comment:
why do these methods take the assorted components of an eventKey struct instead of an actual eventKey?  I would *want* my callers to understand that they're providing a key, not just random args.  I see this in several places here so I guess I'm missing something, but I would have called this an anti-pattern.

Reply:
eventKey is private to the package, so it would have to be made public. That's easy enough. But that relates to my original concern: the event key isn't sufficient to load the event, you need the branch token also. So that would still have to be separate.

I think packing the args that do go into the key into a struct is reasonable though. I'm happy to make that change. Anyone else have thoughts?